### PR TITLE
allow users to use marketIds as symbols in missing functions

### PR DESCRIPTION
**Pull request by pcriadoperez**
We allow users to send in marketIds as symbols, however this had not been implemented yet in all functions.

This allows to use marketIds as symbols for those functions missing the functionality.

In some cases it also moves the call of `this.marketSymbols()` to before the endpoint is called to fail faster.

### CODE REVIEW

**Comment by carlosmiei** [comment]
@pcriadoperez I'm seeing a lot of this pattern

```
symbols = this.marketSymbols (symbols);
// code
this.filterByArray (tickers, 'symbol', symbols);
```

Do you think that it could be useful to have a function called `filterBySymbols` that would sanitize symbols internally? so that devs would not have to worry about it

Something like this:

```Javascript
filterBySymbols (objects, symbols, indexed = true) {
    symbols = this.marketSymbols (symbols);
     return this.filterByArray (objects, 'symbol', symbols, indexed);
}
```

What do you think?

**Comment by pcriadoperez** [comment]
> @pcriadoperez I'm seeing a lot of this pattern
> 
> ```js
> symbols = this.marketSymbols (symbols);
> // code
> this.filterByArray (tickers, 'symbol', symbols);
> ```
> 
> Do you think that it could be useful to have a function called `filterBySymbols` that would sanitize symbols internally? so that devs would not have to worry about it
> 
> Something like this:
> 
> ```js
> filterBySymbols (objects, symbols, indexed = true) {
>      symbols = this.marketSymbols (symbols);
>      return this.filterByArray (objects, 'symbol', symbols, indexed);
> }
> ```
> 
> let me know your thoughts on this

It's a good question.

Initially I agree with calling `this.symbol` inside the filter function to protect from a dev not doing it before. With the same logic I opened this PR (https://github.com/ccxt/ccxt/pull/14622)

However I do see the benefit of keeping it separate with the benefit of calling `this.marketSymbols` at the beginning of the method and throwing an error early if there is an incorrect symbol.
Also I think it's good it makes the dev aware that the user can input both marketId and symbol. (I only realized this recently when you pointed it out)

I think it does make sense to have a `this.filterBySymbol` function. This can be useful for users and easier for new devs to find instead of this.filterByArray.

In conclusion not sure what's the best answer but I think I would aim to do both. I believe it's useful to have a function that's this.filterBySymbol() but also try and call this.marketSymbols at the beginning of the method.

**Comment by carlosmiei** [comment]
@pcriadoperez 

> I believe it's useful to have a function that's this.filterBySymbol() but also try and call this.marketSymbols at the beginning of the method.

I agree that it can be didactic explicitly convert ids/symbols->ids. However, I don't think we should repeat that operation twice, one inside the unified method and the other inside filterByX, cause we're adding unnecessary lookups/conversions that impact performance (probably negligible but still, we should aim for a clean and dry code)

**Comment by pcriadoperez** [comment]
@carlosmiei Makes sense. In that case the only two reasons I can think to not to include it in the filter, is if we want it to fail before the endpoint is called if the user inputs a wrong symbol. However probably every endpoint will fail with the wrong symbols so the user will still get the error, just maybe a little slower. The other worry is if there may be partial failures. So for example if a user inputs several symbols, some correct and some incorrect, and then we call an endpoint that works for the correct symbols and not the incorrect ones.

However I think there is a small chance for both these worries. And makes sense to have a filterBySymbol. Should I open a PR for it?

**Comment by carlosmiei** [comment]
@pcriadoperez 
> if we want it to fail before the endpoint is called if the user inputs a wrong symbol. 

I agree we should crash as soon as possible, from the moment we have a  "erroneous state", going to review this shortly

File: js/binance.js [hunk 1]
```diff
@@ -4839,9 +4839,7 @@ module.exports = class binance extends Exchange {
             const parsed = this.parseFundingRate (entry);
             result.push (parsed);
         }
-        if (symbols !== undefined) {
-            symbols = this.marketSymbols (symbols);
-        }
+        symbols = this.marketSymbols (symbols);
```
Comment (carlosmiei) [inline js/binance.js:4842]: Shouldn't we be doing this earlier?

**Review by carlosmiei — APPROVED** [review]
